app_dart/lib/src/request_handlers/append_log.dart

```diff
@@ -3,41 +3,56 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:typed_data';
 
 import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/service/stackdriver_logger.dart';
 import 'package:gcloud/db.dart';
 import 'package:meta/meta.dart';
 
 import '../datastore/cocoon_config.dart';
 import '../model/appengine/key_helper.dart';
 import '../model/appengine/log_chunk.dart';
+import '../model/appengine/task.dart';
 import '../request_handling/api_request_handler.dart';
 import '../request_handling/authentication.dart';
 import '../request_handling/body.dart';
 import '../request_handling/exceptions.dart';
 
-@immutable
+// ignore: must_be_immutable
```

Comment:
Why this becomes ignored?

Reply:
Updated to follow the style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#comment-all--ignores

We need to be able to inject `requestBody` for the `AppendLog` tests. Ideally, I would have updated `ApiRequestHandler` (the parent class), but doing so makes it no longer immutable, and removes immutability from all the children. I don't know of a good solution yet, so I have just filed an issue to track progress on improving this: https://github.com/flutter/flutter/issues/46902

Figuring this out means we can get more test coverage on the backend code that uses requestBody as it seems no where else in the code tests when requestBody is used.

Reply:
Instead of injecting `requestBody`, how about using a local http client to send the request with body? Similar to https://github.com/flutter/cocoon/blob/master/app_dart/test/request_handling/api_request_handler_test.dart#L49

Reply:
I made it a constructor injection to get this working. Originally, I was trying to do it with parameter injection. This ignore is no longer needed as I was able to keep everything immutable.